### DISCUSSION TITLE
Converting Multipage TIFF to PNG

### DISCUSSION THREAD

**Discussion by usagibear** [SixLabors/ImageSharp#2377] [Q&A] [answered]
Hello, I am new to using ImageSharp. I am trying to convert a stream that has TIFF data and convert each frame into a PNG.

Is there a next frame or something like that or some way to iterate through or select the current frame?

Currently, this is what I have for converting a local TIFF into PNGs. Currently, only the first image/frame is saved but there may be multiple frames.

```c#
using (Image image = Image.Load("test.tif"))
{
    Console.WriteLine($"{image}");
    Console.WriteLine($"image frames: {image.Frames}");
    Console.WriteLine($"image frames count: {image.Frames.Count}");

    for (int i = 0; i < image.Frames.Count; i++)
    {
        Console.WriteLine($"{image.Frames[i]}");


        image.SaveAsPng($"test{i}.png");
    }
}
``` 

When I switch to using streams, I assume I would use the same way of loading (since `Load` is an overloaded function that has a stream variant).

~~Also, is there a way to convert TIFFs to PDFs instead of PNGs with ImageSharp?~~ PDF is not an image format so is not supported.

Thanks!

**Comment by JimBobSquarePants** [comment, marked as answer]
PDF (Portable Document Format) is not an image format. As such it is not supported by the library. You would need a PDF library that can import images.

**Reply by usagibear** [reply]
Thanks! I will update my question.

**Reply by JimBobSquarePants** [reply]
There's a property on `Image` and `Image<TPixel>` called `Frames`. You can enumerate through that collection and save each frame separately.

**Reply by usagibear** [reply]
I see. There doesn't appear to be a `Save()` method for ImageFrames, what method would I use to save each frame?

**Reply by JimBobSquarePants** [reply]
You need to clone the frame into a new `Image` via [ImageFrameCollection.CloneFrame](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.ImageFrameCollection.html#SixLabors_ImageSharp_ImageFrameCollection_CloneFrame_System_Int32_).

**Reply by usagibear** [reply]
Works like a charm. Thanks for your help JimBobSquarePants!
